Corona-Warn-App/src/main/java/de/rki/coronawarnapp/covidcertificate/common/statecheck/DccStateChecker.kt

```diff
@@ -0,0 +1,49 @@
+package de.rki.coronawarnapp.covidcertificate.common.statecheck
+
+import dagger.Reusable
+import de.rki.coronawarnapp.appconfig.AppConfigProvider
+import de.rki.coronawarnapp.covidcertificate.common.certificate.CwaCovidCertificate
+import de.rki.coronawarnapp.covidcertificate.common.certificate.DccData
+import de.rki.coronawarnapp.covidcertificate.expiration.DccExpirationChecker
+import de.rki.coronawarnapp.covidcertificate.signature.core.DscRepository
+import de.rki.coronawarnapp.covidcertificate.signature.core.DscSignatureValidator
+import de.rki.coronawarnapp.util.TimeStamper
+import kotlinx.coroutines.flow.Flow
+import kotlinx.coroutines.flow.combine
+import timber.log.Timber
+import javax.inject.Inject
+
+@Reusable
+class DccStateChecker @Inject constructor(
+    private val timeStamper: TimeStamper,
+    private val appConfigProvider: AppConfigProvider,
+    private val dscRepository: DscRepository,
+    private val dscSignatureValidator: DscSignatureValidator,
+    private val expirationChecker: DccExpirationChecker,
+) {
+
+    suspend fun checkState(
+        dccData: DccData<*>
+    ): Flow<CwaCovidCertificate.State> = combine(
+        appConfigProvider.currentConfig,
+        dscRepository.dscData
+    ) { appConfig, dscData ->
+        val isSignatureValid = dscSignatureValidator.isSignatureValid(dscData, dccData)
+        if (!isSignatureValid) {
+            Timber.tag(TAG).w("Certificate had invalid signature: %s (kid)", dccData)
+            return@combine CwaCovidCertificate.State.Invalid
+        }
```

Comment:
`DscSignatureValidator` does not return boolean anymore , you should wrap`dscSignatureValidator.isSignatureValid(dscData, dccData)`  in a` try {} catch {}` 
if it throw an `InvalidHealthCertificateException` -> `Invalid` ,otherwise it should be `Valid`

Reply:
Done
https://github.com/corona-warn-app/cwa-app-android/pull/3799/commits/256de94211174800a2c7a04ac3218e26a1e55535